lib/hyacinth/adapters/external_identifier_adapter/README.md

```diff
@@ -0,0 +1,2 @@
+For datacite subsystem info, please see docs/datacite.README.md
```

Comment:
This mostly empty README file that points to another README file seems unnecessary.  @barmintor @JackBlackLight what do you think?

Reply:
@fcd1 and I had different preferences about how to handle this, and compromised on this solution.

Reply:
@barmintor  - The file can be removed if that is preferred option.

Reply:
will do, thanks!